### PR TITLE
Arm64: a slew of warning fixes

### PR DESCRIPTION
**Pull request by ligfx**

### CODE REVIEW

**Comment by lioncash** [comment]
These warning fixes should likely have been within their own commits, so that it's known what warnings exactly were being fixed.

**Comment by ligfx** [comment]
@lioncash Thanks for keeping me honest. Updated.

File: Source/Core/Core/PowerPC/JitArm64/JitArm64_Integer.cpp [hunk 1]
```diff
@@ -115,6 +115,21 @@ void JitArm64::reg_imm(u32 d, u32 a, u32 value, u32 (*do_op)(u32, u32),
   }
 }
 
+u32 bit_or(u32 a, u32 b)
```
Comment (lioncash) [inline Source/Core/Core/PowerPC/JitArm64/JitArm64_Integer.cpp]: These can be `static constexpr`. They should likely be named `BitOR`, `BitAND`, `BitXOR` to follow the coding style (ideally, [std::bit_or](http://en.cppreference.com/w/cpp/utility/functional/bit_or), [std::bit_and](http://en.cppreference.com/w/cpp/utility/functional/bit_and), and [std::bit_xor](http://en.cppreference.com/w/cpp/utility/functional/bit_xor) helpers would be used, but it's a pain in non-templated contexts for functor structs).

Comment (ligfx) [inline Source/Core/Core/PowerPC/JitArm64/JitArm64_Integer.cpp]: Good catch, fixed. Yeah I took a look at those functors, it's a shame they're not also provided as actual functions :/

**Pull request merged**